frontend/survey-project/src/app/shared/services/auth.service.ts

```diff
@@ -10,9 +16,47 @@ export class AuthService {
 
     constructor(
         private readonly cookieService: CookieService,
+        private readonly router: Router,
+        private readonly userService: UserService
+
     ) {}
 
     loginUser(session: Session): void {
         this.cookieService.set('ssid', session.id, new Date(session.ending_time));
     }
+
+    logoutUser(): void {
+    
+        this.userService.logoutUser().subscribe({
+            next: () => {
+                this.cookieService.delete('ssid'); 
+                this.router.navigate([RouteNames.LoginRoute]); 
+
+            },
+            error: (err) => {
+                console.error('Error during logout:', err);
+            }
+        });
+    }
+    
+    
+
+    goToRegister(): void {
```

Comment:
Change the name of the method, this is basically something like onDeleteUserAcount.